libpromises/policy.c

```diff
@@ -388,6 +408,38 @@ Policy *PolicyMerge(Policy *a, Policy *b)
         bdp->parent_policy = result;
     }
 
+    StringMap *extra_hashes = NULL;
+    if (a->policy_files_hashes != NULL)
+    {
+        result->policy_files_hashes = a->policy_files_hashes;
+        a->policy_files_hashes = NULL;
+        extra_hashes = b->policy_files_hashes;
```

Comment:
You destroy `extra_hashes` further down. Shouldn't you set this to `NULL` then?

Reply:
Yeah, I should. Although, `a` and `b` are just free'd not, `PolicyDestroy()`'d so it doesn't really matter.